File: prid1920-g01/Models/Tag.cs
using System;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.ComponentModel.DataAnnotations.Schema;


namespace prid1920_g01.Models
{

    public class Tag
    {

        [Key]
        public int Id { get; set; }
        [Required(ErrorMessage = "Required Name")]
        public string Name { get; set; }
        public virtual IList<PostTag> PostTags { get; set; } = new List<PostTag>();

        [NotMapped]
        public IEnumerable<Post> Posts { get => PostTags.Select(t => t.Post); }

    }
}

File: prid1920-g01/ClientApp/src/app/models/tag.ts
import { List } from "lodash";
import { Post } from "./post";

export class Tag{
    id: string;
    name: string;
    posts: List<Post>;

    constructor(data: any){
        this.id = data.id;
        this.name = data.name;
        this.posts = data.posts;
    }
}